reports/hosted-services/main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"log"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/ministryofjustice/cloud-platform-environments/pkg/authenticate"
+	"github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/reports/pkg/hoodaw"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/networking/v1beta1"
+	"k8s.io/client-go/kubernetes"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// resourceMap is used to store both string:string and string:map[string]string key
+// value pairs. The HOODAW API requires the first entry of map to contain a string:string,
+// the rest of the map consists of a primary key (string) with a value containing a map (string:string)
+// of Kubernetes namespaces and hostnames that don't contain the annotation contained within
+// the variable 'annotation'.
+type resourceMap map[string]interface{}
+
+// Namespace describes a Cloud Platform namespace object.
+type namespace struct {
+	Application      string
+	BusinessUnit     string
+	DeploymentType   string
+	Cluster          string
+	DomainNames      []string
+	GithubURL        string
+	Name             string
+	RbacTeam         []string
+	TeamName         string
+	TeamSlackChannel string
+}
+
+// // AllNamespaces contains the list of namespaces of type Namespace.
+// type allNamespaces struct {
+// 	Namespaces []namespace
+// }
+
+var (
+	bucket         = flag.String("bucket", os.Getenv("KUBECONFIG_S3_BUCKET"), "AWS S3 bucket for kubeconfig")
+	ctx            = flag.String("context", "live.cloud-platform.service.justice.gov.uk", "Kubernetes context specified in kubeconfig")
+	hoodawApiKey   = flag.String("hoodawAPIKey", os.Getenv("HOODAW_API_KEY"), "API key to post data to the 'How out of date are we' API")
+	hoodawEndpoint = flag.String("hoodawEndpoint", "/hosted_services", "Endpoint to send the data to")
+	hoodawHost     = flag.String("hoodawHost", os.Getenv("HOODAW_HOST"), "Hostname of the 'How out of date are we' API")
+	kubeconfig     = flag.String("kubeconfig", "kubeconfig", "Name of kubeconfig file in S3 bucket")
+	region         = flag.String("region", os.Getenv("AWS_REGION"), "AWS Region")
+
+	endPoint = *hoodawHost + *hoodawEndpoint
+)
+
+func main() {
+	flag.Parse()
+
+	// Gain access to a Kubernetes cluster using a config file stored in an S3 bucket.
+	clientset, err := authenticate.CreateClientFromS3Bucket(*bucket, *kubeconfig, *region, *ctx)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	namespaces, _ := GetNamespaces(clientset)
```

Comment:
Any reason why we're ignoring the error here?

Reply:
Thanks @jasonBirchall. Missed to catch the err returned. Added a commit to fix that.